Mathlib/CategoryTheory/Enriched/Opposite.lean

```diff
@@ -0,0 +1,160 @@
+--  • the opposite category of the underlying category of `C`.
+-- We also show that if `C` is an enriched ordinary category (i.e. a category enriched in `V`
+-- equipped with an identification `(X ⟶ Y) ≃ (𝟙_ V ⟶ (X ⟶[V] Y))`) then `Cᵒᵖ` is again
+-- an enriched ordinary category.
+section
+
+variable (C : Type u) [EnrichedCategory V C]
+
+/-- The functor going from the underlying category of `Cᵒᵖ` to the opposite of the underlying
+category of `C`. -/
+def ForgetEnrichment.Opposite.toOp : ForgetEnrichment V Cᵒᵖ ⥤ (ForgetEnrichment V C)ᵒᵖ where
+  obj x := x
+  map {x y} f := f.op
+  map_comp {x y z} f g := by
+    have : (f ≫ g) = homTo V (f ≫ g) := rfl
+    rw [this, forgetEnrichment_comp, Category.assoc, tensorHom_eComp_op_eq,
+      leftUnitor_inv_braiding_assoc, ← unitors_inv_equal, ← Category.assoc]
+    congr 1
+
+/-- The functor going from the opposite of the underlying category of `C` to the underlying
+category of `Cᵒᵖ`. -/
+def ForgetEnrichment.Opposite.fromOp : (ForgetEnrichment V C)ᵒᵖ ⥤ ForgetEnrichment V Cᵒᵖ where
+  obj x := x
+  map {x y} f := f.unop
+  map_comp {x y z} f g := by
+    have : g.unop ≫ f.unop = homTo V (g.unop ≫ f.unop) := rfl
+    dsimp
+    rw [this, forgetEnrichment_comp, Category.assoc, unitors_inv_equal,
+      ← leftUnitor_inv_braiding_assoc]
+    have : (β_ _ _).hom ≫ (homTo V g.unop ⊗ homTo V f.unop) ≫
+        eComp V («to» V z.unop) («to» V y.unop) («to» V x.unop) =
+        ((homTo V f.unop) ⊗ (homTo V g.unop)) ≫ eComp V x y z :=
+      by exact (tensorHom_eComp_op_eq V _ _).symm
+    rw [this, ← Category.assoc]
+    congr 1
+
+/-- The identity functor on `ForgetEnrichment V (Cᵒᵖ)` is naturally isomorphic to the composite
+functor from `ForgetEnrichment V (Cᵒᵖ)` to `(ForgetEnrichment V C)ᵒᵖ` and
+back to `ForgetEnrichment V (Cᵒᵖ)`. -/
+def ForgetEnrichment.Opposite.toFromNatIso :
+    𝟭 (ForgetEnrichment V Cᵒᵖ) ≅ toOp V C ⋙ fromOp V C where
+  hom := {app := 𝟙}
+  inv := {
+    app := 𝟙
+    naturality := fun ⦃X Y⦄ f => by
+      dsimp
+      have :  𝟙 ((fromOp V C).obj ((toOp V C).obj Y)) = 𝟙 Y := rfl
+      rw [← this, Category.comp_id, Category.id_comp f]
+      congr 1
+  }
+
+/-- The composite functor from `(ForgetEnrichment V C)ᵒᵖ` to `ForgetEnrichment V (Cᵒᵖ)` and
+back to `(ForgetEnrichment V C)ᵒᵖ` is naturally isomorphic to the identity functor. -/
+def ForgetEnrichment.Opposite.fromToNatIso :
+    fromOp V C ⋙ toOp V C ≅ 𝟭 (ForgetEnrichment V C)ᵒᵖ where
+  hom := {
+    app := 𝟙
+    naturality := fun ⦃X Y⦄ f => by
+      dsimp
+      have :  𝟙 ((toOp V C).obj ((fromOp V C).obj Y)) = 𝟙 Y := rfl
+      rw [← this, Category.comp_id, Category.id_comp f]
+      congr 1
+  }
+  inv := {app := 𝟙}
+
+/-- The equivalence between the underlying category of `Cᵒᵖ` and the opposite of the underlying
+category of `C`. -/
+def ForgetEnrichment.Opposite.equiv : ForgetEnrichment V Cᵒᵖ ≌ (ForgetEnrichment V C)ᵒᵖ :=
+  Equivalence.mk (toOp V C) (fromOp V C) (toFromNatIso V C) (fromToNatIso V C)
```

Comment:
When constructing isomorphisms, you may use `NatIso.ofComponents`, which requires checking the naturality only in one direction, which can be achieved automatically here:
```suggestion
@[simps]
def forgetEnrichmentOppositeEquivalence :
    ForgetEnrichment V Cᵒᵖ ≌ (ForgetEnrichment V C)ᵒᵖ where
  functor := ForgetEnrichment.Opposite.toOp V C
  inverse := ForgetEnrichment.Opposite.fromOp V C
  unitIso := NatIso.ofComponents (fun _ ↦ Iso.refl _)
  counitIso := NatIso.ofComponents (fun _ ↦ Iso.refl _)
```
Also, `Equivalence.mk` is not recommended (unless we are only interested in the existence of an abstract equivalence), because that constructor does not require the additional compatibility `functor_unitIso_comp` (which is obtained automatically here!) which is also satisfied by adjunction and changes one of the isos in order to get this additional compatibility.
I would suggest using the name `forgetEnrichmentOppositeEquivalence`, and renaming `ForgetEnrichment.Opposite.toOp` as `forgetEnrichmentOppositeEquivalence.functor` and `fromOp` as `forgetEnrichmentOppositeEquivalence.inverse`.